### PR TITLE
amp-viewer-integration support string messages from webview

### LINKED ISSUE #8182: amp-viewer-integration webview messages are passed as strings
webview messages are passed as strings and this needs to be supported in the amp-viewer-integration

### PR DESCRIPTION
**Pull request by chenshay**
amp-viewer-integration:
The messages from the webview arrive as strings. Support this.

close #8182

### CODE REVIEW

File: extensions/amp-viewer-integration/0.1/messaging.js [hunk 1]
```diff
@@ -156,12 +159,11 @@ export class Messaging {
    */
   handleMessage_(event) {
     dev().fine(TAG, 'AMPDOC got a message:', event.type, event.data);
-    /** @type {Message} */
-    const message = event.data;
+    const message = this.isWebview_ ? JSON.parse(event.data) : event.data;
     if (message.type == MessageType.REQUEST) {
-      this.handleRequest_(message);
+      this.handleRequest_(/** @type {Message} */(message));
```
Comment (dvoytenko) [inline extensions/amp-viewer-integration/0.1/messaging.js]: Move cast above to `const message = ...`

File: examples/viewer-integr-messaging.js [hunk 2]
```diff
@@ -31,18 +31,19 @@ var APP = '__AMPHTML__';
  * https://developer.mozilla.org/en-US/docs/Web/API/Channel_Messaging_API
  */
 class WindowPortEmulator {
-  constructor(messageHandlers, id, port) {
+  constructor(messageHandlers, id, port, opt_isWebview) {
```
Comment (dvoytenko) [inline examples/viewer-integr-messaging.js]: The actual code below is preferred. There you only use pass `isWebview` to the `Messaging` class. It'd be better here as well.

File: extensions/amp-viewer-integration/0.1/messaging.js (hunk 1 above)
Comment (chenshay) [inline extensions/amp-viewer-integration/0.1/messaging.js]: done

File: examples/viewer-integr-messaging.js (hunk 2 above)
Comment (chenshay) [inline examples/viewer-integr-messaging.js]: done

**Pull request merged**